addons/account/tests/test_account_move_in_invoice.py

```diff
@@ -2037,3 +2037,132 @@ def test_fiduciary_mode_date_suggestion(self):
+        for counterpart_move_type, counterpart_amount in counterpart_values_list:
+            if counterpart_move_type == 'payment':
+                create_payment(move, counterpart_amount)
+            elif counterpart_move_type == 'reverse':
+                create_reverse(move, counterpart_amount)
+            else:
+                counterpart_move = create_counterpart_move(counterpart_move_type, counterpart_amount)
+                counterpart_line = counterpart_move.line_ids.filtered(lambda x: x.account_id == line.account_id)
+                (line + counterpart_line).reconcile()
+
+        if payment_state == 'in_payment' and move._get_invoice_in_payment_state() == 'paid':
+            payment_state = 'paid'
+
+        self.assertRecordValues(move, [{'payment_state': payment_state}])
+
+    def test_payment_move_state(self):
+        for move_type, amount, counterpart_values_list, payment_state in (
+            ('out_invoice', 1000.0, [('out_refund', 1000.0)], 'reversed'),
+            ('out_invoice', 1000.0, [('reverse', 1000.0)], 'reversed'),
+            ('out_receipt', 1000.0, [('out_refund', 1000.0)], 'reversed'),
+            ('out_receipt', 1000.0, [('reverse', 1000.0)], 'reversed'),
+            ('out_refund', 1000.0, [('reverse', -1000.0)], 'reversed'),
+            ('in_invoice', 1000.0, [('in_refund', 1000.0)], 'reversed'),
+            ('in_invoice', 1000.0, [('reverse', 1000.0)], 'reversed'),
+            ('in_invoice', 1000.0, [('in_refund', 1000.0)], 'reversed'),
+            ('in_invoice', 1000.0, [('reverse', 1000.0)], 'reversed'),
+            ('in_refund', 1000.0, [('reverse', 1000.0)], 'reversed'),
+            # ('entry', 1000.0, [('entry', 1000.0)], '???'),
+            # ('entry', 1000.0, [('reverse', 1000.0)], 'reversed'),
+
+            ('out_invoice', 1000.0, [('payment', 500.0)], 'partial'),
+            ('out_invoice', 1000.0, [('payment', 1000.0)], 'in_payment'),
+            ('out_receipt', 1000.0, [('payment', 500.0)], 'partial'),
+            ('out_receipt', 1000.0, [('payment', 1000.0)], 'in_payment'),
+            ('out_refund', 1000.0, [('payment', 500.0)], 'partial'),
+            ('out_refund', 1000.0, [('payment', 1000.0)], 'in_payment'),
+            ('in_invoice', 1000.0, [('payment', 500.0)], 'partial'),
+            ('in_invoice', 1000.0, [('payment', 1000.0)], 'in_payment'),
+            ('in_receipt', 1000.0, [('payment', 500.0)], 'partial'),
+            ('in_receipt', 1000.0, [('payment', 1000.0)], 'in_payment'),
+            ('in_refund', 1000.0, [('payment', 500.0)], 'partial'),
+            ('in_refund', 1000.0, [('payment', 1000.0)], 'in_payment'),
+            # ('entry', 1000.0, [('payment', 500.0)], 'partial'),
+            # ('entry', 1000.0, [('payment', 1000.0)], 'paid'),
+
+            ('out_invoice', 1000.0, [('out_refund', 500.0), ('payment', 500.0)], 'in_payment'),
```

Comment:
test
- partial refund only (in and out) IMO that should be 'not_paid'
- in_invoice with part payment & part refund
- out_in with part invoice, part refund, (not fully reconciled)